index.js

```diff
@@ -47,6 +47,54 @@ const Dialog = props => {
   const { id, classNames } = props
   const titleId = props.titleId || id + '-title'
   const Element = props.useDialog ? 'dialog' : 'div'
+  const CloseButton = () => {
+    return (
+      <button
+        type='button'
+        aria-label={props.closeButtonLabel}
+        onClick={close}
+        className={classNames.closeButton}
+      >
+        {props.closeButtonContent}
+      </button>
+    )
+  }
+  const Heading = () => {
+    /**
+     * Using a paragraph with accessibility mapping to work around SEO
+     * concerns of having multiple <h1> per page.
+     * See: https://twitter.com/goetsu/status/1261253532315004930
+     */
+    return (
+      <p
+        id={titleId}
+        className={classNames.title}
+        role='heading'
+        aria-level='1'
+      >
+        {props.title}
+      </p>
+    )
+  }
+  const getChildOrder = () => {
+    if (props.closeButtonPosition === 'first') {
+      return (
+        <React.Fragment>
+          <CloseButton />
+          <Heading />
+        </React.Fragment>
+      )
+    } else if (props.closeButtonPosition === 'last') {
```

Comment:
So this is not technically correct. If `first`, it should be “button + heading + children”. But if `last`, it should be “heading + children + button”.